### ISSUE TITLE
Panic in RSA verification on real-world data

### ISSUE THREAD

**Issue by Shnatsel** [briansmith/ring#929]
I have been stress-testing `ureq` by downloading the top million websites with it and encountered a panic in `ring::limb::big_endian_from_limbs`. Since it appears to be a private function (does not show up on docs.rs) I am reporting the bug here.

Steps to reproduce:
Run [this simple program](https://gist.githubusercontent.com/deltaphc/2949ed292c7d1169e744e5ffa7fd0687/raw/bb5e5f30bbfa38406a6bd7bd538224f450717db3/ureq_test.rs) with "sitios.cl" given as the only command-line argument.

Backtrace:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `389`,
 right: `392`', /home/somebody/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.9/src/limb.rs:245:5
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1057
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:195
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:215
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:472
  11: rust_begin_unwind
             at src/libstd/panicking.rs:376
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:84
  13: ring::limb::big_endian_from_limbs
  14: ring::rsa::verification::verify_rsa_
  15: ring::rsa::verification::<impl ring::signature::VerificationAlgorithm for ring::rsa::RsaParameters>::verify
  16: ring::signature::UnparsedPublicKey<B>::verify
  17: webpki::signed_data::verify_signature
  18: webpki::EndEntityCert::verify_signature
  19: rustls::verify::verify_signed_struct
  20: <rustls::client::tls12::ExpectServerDone as rustls::client::hs::State>::handle
  21: <rustls::client::tls12::ExpectServerDoneOrCertReq as rustls::client::hs::State>::handle
  22: rustls::client::ClientSessionImpl::process_main_protocol
  23: <rustls::client::ClientSession as rustls::session::Session>::process_new_packets
  24: rustls::session::Session::complete_io
  25: rustls::stream::Stream<S,T>::complete_prior_io
  26: <rustls::stream::StreamOwned<S,T> as std::io::Write>::write
  27: std::io::Write::write_all
  28: ureq::unit::connect
  29: ureq::unit::connect
  30: ureq::request::Request::do_call
  31: ureq::request::Request::call
  32: ureq_test::main
  33: std::rt::lang_start::{{closure}}
  34: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  35: std::panicking::try::do_call
             at src/libstd/panicking.rs:296
  36: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:79
  37: std::panicking::try
             at src/libstd/panicking.rs:272
  38: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  39: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  40: std::rt::lang_start
  41: __libc_start_main
  42: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

This is reproducible with a total of 7 websites out of the [majestic million](https://blog.majestic.com/development/majestic-million-csv-daily/): adriatic.hr, ascii.cl, doctorsonly.co.il, elektroshopwagner.de, hbo-kennisbank.nl, josefsson.org, neteasy.pl, sitios.cl

**Comment by andrewtj** [comment]
Looks like the same panic I ran into [spiking out 1024-bit RSA](https://github.com/briansmith/ring/issues/870#issuecomment-510410483).

**Comment by Shnatsel** [comment]
This panic is also reachable by running https://github.com/neonmoe/minreq on real-world websites

**Comment by vavrusa** [comment]
I think this is fixed here https://github.com/briansmith/ring/pull/871#issuecomment-521359182

**Comment by briansmith** [comment]
Thanks everybody. I put up a draft PR for fixing this: https://github.com/briansmith/ring/pull/937. I'm going to work on the tests tomorrow.

**Comment by briansmith** [comment]
Fixed in #937 in *ring* 0.16.10.
